start_control_ssm.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+log() {
+  echo "$*" >&2
+}
+
+enable_hybrid_env_ssm() {
+  # SSM parameters necessary to register with a hybrid activation
+  activation_code=$(get_user_data_ssm "activation-code")
+  activation_id=$(get_user_data_ssm "activation-id")
+  region=$(get_user_data_ssm "region")
+
+  # Register with AWS Systems Manager (SSM)
+  amazon-ssm-agent -register -code "${activation_code}" -id "${activation_id}" -region "${region}"
+  if [[ "$?" -eq 1 ]]; then
+    log "Failed to register with AWS Systems Manager (SSM)"
+    exit 1
+  fi
+}
+
+# Extract the values from user-data json, and exit on failure (if any)
+get_user_data_ssm() {
+  local param_value
+  local ssm_param="${1:?}"
```

Comment:
Might be good to take the file as a parameter too, rather than assuming a global USER_DATA, so that the function is easier to understand on its own.

I'm not sure this ssm_param argument is carrying its weight - it seems simpler and clearer to me to have a general `get_json` (or whatever) function that you pass `.ssm.activation-code` instead of assuming the `.ssm` prefix and doing the jq variable magic to stitch them together.